### ISSUE TITLE
Video embeds

### ISSUE THREAD

**Issue by MMMaellon** [oleeskild/obsidian-digital-garden#80]
I would love to be able to embed videos in my notes and have them appear on my digital garden as well. Right now, it just shows `![[videofilename.mp4]]` in it's place instead.

**Comment by oleeskild** [comment]
Thanks for the suggestion! This could perhaps be something to look into in the future. Currently the way pictures are uploaded is the whole image is embedded right in the note. (If you look at a note with an image in your repo you will see that there is a really long text string in the note. This is the entire picture encoded as text). Doing the same for video isn't really feasible. If I, at a later point, implement some sort of CDN support to upload large files, this could perhaps be something to look into. 

What you could do right now to make this possible is to use the embed feature of youtube videos. This would create a youtube player in your note. (This should work both in obsidian and in your garden). It would look something like this:

```
<iframe width="560" height="315" src="https://www.youtube.com/embed/djV11Xbc914" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```

The downside of course is that you would have to upload the video to youtube first.

**Comment by stalegjelsten** [comment]
I would love for Digital garden to support the YouTube [embed syntax used in Obsidian](https://help.obsidian.md/Editing+and+formatting/Embedding+web+pages#Embed+a+YouTube+video): `![alt text](https://youtube.com/watch?v=VIDEO_ID)`.

At the moment, this syntax is always interpreted as an image by Digital garden, with resulting `<img src="https://youtube.com/watch?v=VIDEO_ID" alt="alt text">.

**Comment by julesvirallinen** [comment]
MVP using markdown-it-video: 

plugin: https://github.com/julesvirallinen/obsidian-digital-garden/pull/10
template: https://github.com/julesvirallinen/digitalgarden-dev/pull/1

<img width="745" alt="CleanShot 2023-09-19 at 13 41 54@2x" src="https://github.com/oleeskild/obsidian-digital-garden/assets/4622905/ec02e1aa-cf4c-430b-9bdb-62960ce79217">

**Comment by jsifalda** [comment]
any updates?

**Comment by stalegjelsten** [comment]
@jsifalda for youtube embeds I'm using a custom filter in the Obsidian Digital Garden settings. It replaces all occurences of ![](https://youtube.com/watch?v=VIDEO_ID) with an iframe video embed.

Search for
`!\[(.*)\]\(https?:\/\/w*\.?(?:(?:youtu\.be\/)|(?:youtube\.com\/watch\?v=))(.*)\)` 

Replace with
`<iframe src="https://www.youtube.com/embed/$2" class="youtube" title="$1" loading="lazy" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>`

**Comment by jsifalda** [comment]
@stalegjelsten thanks, it is the very clever solution!
do you have also a similar regex for twitter/x please? 🙏

**Comment by jsifalda** [comment]
🚨 i found the way for twitter/x.com as well, guide is here https://www.craftengineer.com/embedding-x-twitter-links-in-your-obsidian-digital-garden/ 😇

**Comment by stalegjelsten** [comment]
Great post @jsifalda, but your filter is working too well because it's impossible to see the original Twitter syntax since it's been replaced by HTML. 😅 
<img width="738" alt="bilde" src="https://github.com/oleeskild/obsidian-digital-garden/assets/5513050/c14b0e79-184c-4f8b-a557-08df83b28593">

**Comment by jsifalda** [comment]
that is unfortunate, the same for the yt as well tho 😂

**Comment by jsifalda** [comment]
anyway it is fixed now ;) 
thanks for heads-up @stalegjelsten

**Comment by kerim** [comment]
I am not very good with regex, but with the help of ChatGPT I've updated the filter so it can handle all three kinds of YouTube links: youtu.be/, youtube.com/watch?v=, or youtube.com/embed/:

```
!\[(.*?)\]\(https?:\/\/w*\.?(?:youtu\.be\/|youtube\.com\/watch\?v=|youtube\.com\/embed\/)(.*?)\)
```

```
<iframe src="https://www.youtube.com/embed/$2" title="$1" style="width:100%; aspect-ratio:16/9" loading="lazy" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
```

**Comment by GJRobert** [comment]
This feature would be great~~~

**Comment by mpresecan** [comment]
Yes, this is very anticipated feature. `![video](https://www.youtube.com/watch?v=dQw4w9WgXcQ)` would do much better, than pasting iframe

**Comment by jsifalda** [comment]
@GJRobert / @mpresecan this feature exists via custom filters, check this out https://www.craftengineer.com/embedding-x-twitter-links-in-your-obsidian-digital-garden/

**Comment by mpresecan** [comment]
> @GJRobert / @mpresecan this feature exists via custom filters, check this out https://www.craftengineer.com/embedding-x-twitter-links-in-your-obsidian-digital-garden/

Thank you, this is awesome

**Comment by GJRobert** [comment]
> @GJRobert / @mpresecan this feature exists via custom filters, check this out https://www.craftengineer.com/embedding-x-twitter-links-in-your-obsidian-digital-garden/

Merci, vous êtes un génial! Thank you you're a genius!

**Comment by jsifalda** [comment]
@GJRobert / @mpresecan let me know if you are looking for any other tips regarding digital garden or obsidian generally  ;)

**Comment by jsifalda** [comment]
@GJRobert / @mpresecan what are your digital gardens? :) what about exchanging backlinks? 😇

**Comment by GJRobert** [comment]
> @GJRobert / @mpresecan what are your digital gardens? :) what about exchanging backlinks? 😇

Hi @jsifalda I have two DGs currently, both generated from the same Obs vault :)
I have written an crude instruction on my IT DG: [Multiple DGs with 1 vault](https://imazingrace.vercel.app/Obsidian/DG/multiple)

So that is my IT DG. And here's my DG on languages and future of Taiwan: [🏡 Aiuanyu](https://aiuanyu.net/)
Both of them are written mostly in the Chinese language, so you may have to use Web translation tool to skim them 😃

**Comment by jsifalda** [comment]
cool, where can i reach out to you? :)

**Comment by GJRobert** [comment]
> cool, where can i reach out to you? :)

Hi, I have seen that you have followed my Twitter account https://twitter.com/Aiuanyu
You can also find "Aiuanyu" on Facebook, IG, and Threads.
Feel free to talk! Just that there are very few tweets / thread yet. But I'm willing to exchange information.

**Comment by jsifalda** [comment]
🧠 
now, it is also possible to embed https://bsky.app/ links (the same way as yt or x.com), here is a short "how-to" guide: [https://www.craftengineer.com/embedding-bluesky-links-in-your-obsidian-digital-garden/](https://dub.sh/JlfO0oI)

**Comment by foxblock** [comment]
I see the regex replace solutions for embedding YouTube video links.

But coming back to the original questions, these solutions will not work for .mp4 files from your vault.

> Thanks for the suggestion! This could perhaps be something to look into in the future. Currently the way pictures are uploaded is the whole image is embedded right in the note. (If you look at a note with an image in your repo you will see that there is a really long text string in the note. This is the entire picture encoded as text). Doing the same for video isn't really feasible. If I, at a later point, implement some sort of CDN support to upload large files, this could perhaps be something to look into.

I see this is no longer the case (images are uploaded and then linked in the img normally). So maybe the video feature could be revisited? It should work basically the same, but using a video tag instead.
